### ISSUE TITLE
Support delimiter option for "Opening a csv"

### ISSUE THREAD

**Issue by dokutoshi** [microsoft/vscode-data-wrangler#66]
## Environment data

Version: 1.80.0-insider (Universal)
Commit: 39efe30bfc8e196f5be2119479092bd9eb83e174
Date: 2023-06-30T14:34:06.522Z
Electron: 22.3.14
ElectronBuildId: 21893604
Chromium: 108.0.5359.215
Node.js: 16.17.1
V8: 10.8.168.25-electron.0
OS: Darwin x64 22.5.0

## Expected behaviour

Open csv in Data Wrangler parse semicolon as a delimiter

## Actual behaviour

Values are not parsed on semicolon as this is not an option.

## Steps to reproduce:

[**NOTE**: Self-contained, minimal reproducing code samples are **extremely** helpful and will expedite addressing your issue]

1. Use a UCI dataset that contains semicolons -> https://archive.ics.uci.edu/dataset/186/wine+quality

<img width="1188" alt="image" src="https://github.com/microsoft/vscode-data-wrangler/assets/10182821/7cf42c8f-c627-485d-b9a1-9a49a846bb5b">

## Logs

<details>

<summary>Output for <code>Jupyter</code> in the <code>Output</code> panel (<code>View</code>→<code>Output</code>, change the drop-down the upper-right of the <code>Output</code> panel to <code>Jupyter</code>)
</summary>

<p>

```
XXX
```

</p>
</details>

**Comment by pwang347** [comment]
@dokutoshi Thank you for opening this request! This is a feature that we have considered and is on our future roadmap. For a temporary workaround, please see the one mentioned in this issue: https://github.com/microsoft/vscode-data-wrangler/issues/42

**Comment by Rokis1990** [comment]
I had trouble opening a tab delimited CSV file too. I wonder if it's for the same reason.

**Comment by LouisLeNezet** [comment]
I'm stuck with the same problem.
The separator configuration would be a really nice feature.

**Comment by kycutler** [comment]
Hey folks, good news!

We have added support for automatic delimiter detection in the latest pre-release version of Data Wrangler. This should enable opening CSV files with any delimiter and for it to "just work".

We welcome any additional feedback you might have. We still intend to support full customization of loading options in the future. But we hope this unblocks you from using Data Wrangler with non-comma delimited files.

Thanks for trying Data Wrangler!

**Comment by PabloLION** [comment]
@kycutler Seems it does not "just work" for version `v0.23.2`. I opened a file with comma separator (`,`), but the detected separator seems to be a case-insensitive `d` / `D`. 

- This also happed in `v0.22`.
- I also tried with semicolon separator (`;`), the default separator of mac "numbers" export format, but both versions yield the same result.
- I also use "Edit csv" (id: `janisdd.vscode-edit-csv`) and "Rainbow CSV" (id: `mechatroner.rainbow-csv`) and both of them could parse both separators.

should I open a new issue for this?
For me adding a delimiter option is a good work around: I'd prefer to input one character than missing all the functionalities that Data Wrangler offers.

**Comment by hodgigre** [comment]
An ability to edit the read_csv for any number of reasons would be helpful.  Is there a reason it can't edited like any other step in DW?  In addition to changing the delimiter, I could use the ability to specify header=None and the column names I would like to use.  In the interim, as suggested in the other issue a workaround is to author your own read_csv command in Jupyter and then load the resulting df in DW.

**Comment by kycutler** [comment]
@PabloLION would you be able to share the file (or a portion of it) where the auto-detected separator is `d`? That would help us to fine-tune the detection. Also note that in the latest Data Wrangler pre-release (`0.23.3`) you can disable the auto-detection via the Data Wrangler extension settings to unblock opening comma-separated CSV files in cases where the delimiter is not detected correctly.

We are working on allowing the import step to be edited just like other operations as @hodgigre suggests. It's a bit tricky due to some assumptions we currently make, but please keep an eye out for that ability in the not-too-distant future as it will enable several highly requested features such as this one 🙂 I'll update here when that is available as well.

Thank you both for the feedback and please don't hesitate to share if you have more!

**Comment by hodgigre** [comment]
Thanks for the update Kyle. The work and the communication are both appreciated.


________________________________

**Comment by kycutler** [comment]
Hey folks, good news! Editing the import step is now possible in the latest pre-release version of Data Wrangler. After opening a file (and before applying any operations), you can now select the first cleaning step to edit it, and then either adjust the parameters from the operations panel or modify the code directly. Currently there are built-in options for the file path and delimiter, and we plan to add more in the future.

Closing this issue as it should now be resolved, but, as always, any additional feedback is welcome!
